lib/galaxy/datatypes/data.py

```diff
@@ -736,14 +736,12 @@ def estimate_file_lines( self, dataset ):
 
     def count_data_lines(self, dataset):
         """
-        Count the number of lines of data in dataset,
-        skipping all blank lines and comments.
+        Count the number of lines of data in dataset.
+        Elimination of blank lines and comments removed per issue 2694.
```

Comment:
I don't think this is probably the right thing to do here, since it'll include comment lines intentionally *not* counted.  Perhaps just relax the 'blank lines' check, and keep comment lines filtered out?  @tnabtaf, was that your intent with the request?